### PR TITLE
Add modulo operator

### PR DESCRIPTION
**Pull request by parthchadha**
Fixes https://github.com/NVIDIA/TensorRT-Incubator/issues/201

### CODE REVIEW

File: tripy/tripy/frontend/trace/ops/binary_elementwise.py [hunk 1]
```diff
@@ -144,6 +145,39 @@ def to_flat_ir(self, inputs, outputs):
             DivideOp.build(inputs, [divide_out])
             # Then apply FloorOp to the result of the division
             FloorOp.build([divide_out], outputs)
+        elif self.kind == BinaryElementwise.Kind.MOD:
```
Comment (yizhuoz004) [inline tripy/tripy/frontend/trace/ops/binary_elementwise.py:148]: We can use `stablehlo.remainder`: https://github.com/openxla/stablehlo/blob/main/docs/spec.md#remainder

Comment (parthchadha) [reply tripy/tripy/frontend/trace/ops/binary_elementwise.py:148]: @yizhuoz004 remainder and modulo are not the same for negative numbers and I think adding conditionals for checking negative number and adjusting for that will be compute intensive compared to the current implementation.

Comment (yizhuoz004) [reply tripy/tripy/frontend/trace/ops/binary_elementwise.py:148]: I see, it confuses me that stablehlo has `remainder` but not `modulo` which is more common...

File: tripy/tripy/frontend/trace/ops/binary_elementwise.py [hunk 2]
```diff
@@ -528,6 +562,73 @@ def __rfloordiv__(self: Union["tripy.Tensor", Any], other: Union["tripy.Tensor",
...
+        :caption: Example
+
+        a = tp.Tensor([4.0, 6.0])
+        b = tp.Tensor([3.0, 4.0])
+        output = a % b
+
+        assert np.array_equal(cp.from_dlpack(output).get(), np.array([1.0, 2.0]))
+    """
+    from tripy.frontend.trace.ops.cast import cast
+    from tripy.common.datatype import int32
```
Comment (yizhuoz004) [inline tripy/tripy/frontend/trace/ops/binary_elementwise.py:594]: imports are not needed.

File: tripy/tripy/frontend/trace/ops/binary_elementwise.py [hunk 3]
```diff
@@ -528,6 +562,73 @@ def __rfloordiv__(self: Union["tripy.Tensor", Any], other: Union["tripy.Tensor",
...
+        :linenos:
+        :caption: Example
+
+        a = tp.Tensor([4.0, 6.0])
+        output = 2 % a
+
+        assert np.array_equal(cp.from_dlpack(output).get(), np.array([2.0, 2.0]))
+    """
+    from tripy.frontend.trace.ops.cast import cast
+    from tripy.common.datatype import int32
```
Comment (yizhuoz004) [inline tripy/tripy/frontend/trace/ops/binary_elementwise.py:627]: Same here

**Review by yizhuoz004 — APPROVED** [review]

**Pull request merged**